crate/utils/src/crypto/hybrid_encryption/encryption.rs

```diff
@@ -76,15 +71,30 @@ impl HybridEncryptionSystem {
 impl EncryptionSystem for HybridEncryptionSystem {
     fn encrypt(&self, request: &Encrypt) -> Result<EncryptResponse, KmipUtilsError> {
         let plaintext = request.data.clone().context("missing plaintext data")?;
-        let mut rng = self.rng.lock().unwrap();
+
+        #[cfg(not(feature = "fips"))]
+        let rng = Arc::new(Mutex::new(CsRng::from_entropy()));
+        #[cfg(not(feature = "fips"))]
+        let mut rng = rng.lock().expect("RNG lock poisoned");
```

Comment:
Generating a CSRNG is costly. You typically do not want to do it at each encryption.
Plus, here you are wrapping the RNG in a mutex and unwrapping it the line after: this makes no sense.

Reply:
I understand that but since an external crate needs an RNG in input and since I cannot pass the openssl one, I figured this would be the best option.
Plus this allows to make a fips build not include "non-fips" RNG.

I'm open to suggestions for how we can handle this ?

Reply:
Is there no FIPS-validated RNG?

The usual thing to do is the one you reversed: accept the RNG as input.